corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js

```diff
@@ -893,6 +896,100 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
     EthiopianDateEntry.prototype = Object.create(EntrySingleAnswer.prototype);
     EthiopianDateEntry.prototype.constructor = EntrySingleAnswer;
 
+    /**
+     * Base class for entry types that involve uploading a file: multimedia and signatures.
+     */
+    function FileEntry(question, options) {
+        var self = this;
+        EntrySingleAnswer.call(this, question, options);
+        self.templateType = 'file';
+        self.xformAction = Const.ANSWER_MEDIA;
+        self.xformParams = function () {
+            return { file: self.file() };
+        };
+
+        self.file = ko.observable();
+
+    }
+    FileEntry.prototype = Object.create(EntrySingleAnswer.prototype);
+    FileEntry.prototype.constructor = EntrySingleAnswer;
+    FileEntry.prototype.onAnswerChange = function (newValue) {
+        var self = this;
+        if (newValue !== Const.NO_ANSWER) {
+            var $input = $('#' + self.entryId);
+            self.answer(newValue.replace(Const.FILE_PREFIX, ""));
+            self.file($input[0].files[0]);
+        } else {
+            self.answer(newValue);
+            self.file(null);
+        }
+        this.question.onchange();
+    };
+
+    /**
+     * Represents an image upload.
+     */
+    function ImageEntry(question, options) {
+        var self = this;
+        FileEntry.call(this, question, options);
+        self.accept = "image/*";
+
+        self.helpText = function () {
+            return gettext("Upload image");
+        };
+
+    }
+    ImageEntry.prototype = Object.create(FileEntry.prototype);
+    ImageEntry.prototype.constructor = FileEntry;
+
+    /**
+     * Represents an audio upload.
+     */
+    function AudioEntry(question, options) {
+        var self = this;
+        FileEntry.call(this, question, options);
+        self.accept = "audio/*";
+
+        self.helpText = function () {
+            return gettext("Upload audio file");
+        };
+
+    }
+    AudioEntry.prototype = Object.create(FileEntry.prototype);
+    AudioEntry.prototype.constructor = FileEntry;
+
+    /**
+     * Represents a video upload.
+     */
+    function VideoEntry(question, options) {
+        var self = this;
+        FileEntry.call(this, question, options);
+        self.accept = "video/*";
+
+        self.helpText = function () {
+            return gettext("Upload video file");
+        };
+
+    }
+    VideoEntry.prototype = Object.create(FileEntry.prototype);
+    VideoEntry.prototype.constructor = FileEntry;
+
+    /**
+     * Represents a signature, which requires the user to upload a signature file.
+     */
+    function SignatureEntry(question, options) {
+        var self = this;
+        FileEntry.call(this, question, options);
+        self.accept = "image/*,.pdf,.doc,.docx";
```

Comment:
@orangejenny A common use case of this feature on Web Apps can be to add docs and it's not clear to me whether app builders should create an Image upload or a Signature question for that. I think the only distinction between Signature question and Image question potentially can be that Signature question provide additional UX to be able to draw the signature. If we are not doing that, It doesn't make sense to have distinction between these two question types and maybe we should still disable signature question type on Web Apps until we implement the drawing part and have Image question accept all of `image/*,.pdf,.doc,.docx` ?

FYI @Robert-Costello

Reply:
Hmm, I agree that it's ambiguous whether uploading a document should use an image or signature question. As implemented, people may use the signature question to upload arbitrary documents, which would then get weird if/when we add an e-signature widget to signature questions. If we continue to disable signature questions on web apps and instead allow images questions to support pdf uploads, people will use image questions for arbitrary documents, which is also a little weird, but less so. Yeah, I think you're right, let's turn off signature questions.

Reply:
Also linking to the related SolTech ticket - they want to upload PDFs that aren't specifically signatures: https://dimagi-dev.atlassian.net/browse/SC-2425

Reply:
Sounds good. Thanks!

@Robert-Costello  will you be able to take this change -
1. Disable Signature question on Web Apps UI 
2. Have image question accept `image/*,.pdf,.doc,.docx`

Reply:
Yes, I can work on this change.